packages/react/menu/src/Menu.tsx

```diff
@@ -45,6 +49,17 @@ type MenuOwnProps = {
 const Menu: React.FC<MenuOwnProps> = (props) => {
   const { open = false, children, onOpenChange } = props;
   const handleOpenChange = useCallbackRef(onOpenChange);
+
+  React.useEffect(() => {
+    const handleMenuClose = (event: Event) => {
+      if (event.defaultPrevented) return;
+      handleOpenChange(false);
+    };
+
+    document.addEventListener(ITEM_SELECT, handleMenuClose);
+    return () => document.removeEventListener(ITEM_SELECT, handleMenuClose);
+  }, [handleOpenChange]);
```

Comment:
Raised this up to take advantage of bubbling and close all open menus on selection.